src/data-workspace/inputs/generic-input.js

```diff
@@ -43,23 +43,41 @@ export const GenericInput = ({
         )
     }
 
+    const formatValue = (value) => {
+        let formattedValue = value
+
+        if (
+            value &&
+            value.trim() &&
+            NUMBER_TYPES.includes(valueType) &&
+            Number.isFinite(Number(value))
+        ) {
+            formattedValue = Number(value).toString()
+        }
+
+        return formattedValue
+    }
+
     const { input, meta } = useField(fieldname, {
         validate: validatorsByValueType[valueType],
         subscription: { value: true, dirty: true, valid: true },
+        format: formatValue,
+        formatOnBlur: true,
     })
 
     const handleBlur = () => {
         const { value } = input
-        const hasEmptySpaces = value && value.trim() === ''
+        const formattedValue = value ? formatValue(value.trim()) : ``
         const { dirty, valid } = meta
-        if (dirty && valid && !hasEmptySpaces && value !== lastSyncedValue) {
-            syncData(value.trim())
+        if (dirty && valid && formattedValue !== lastSyncedValue) {
+            syncData(formattedValue)
         }
     }
 
     return (
         <input
             {...input}
+            value={input.value ?? ''}
```

Comment:
Another good find. I could also reproduce this. Strangely enough this doesn't happen when you remove the `format` function from the options passed to `useField`. And apart from the way you did it, it can also be fixed by adding a `parse` function. I'd argue that's marginally cleaner because we're now preventing the `input.value` from ever becoming `undefined` rather than having to deal with it being `undefined`.

Reply:
I did try using `parse` on `useField` as I would have preferred to use that - but that didn't fix the problem. I think `parse` might only execute on the inputs onChange event